app/adapters/osz_archive.py

```
@@ -0,0 +1,242 @@
+                f"{limits.max_osu_file_count} limit",
+            )
+
+        osu_members: list[OszOsuMember] = []
+        for info in osu_infos:
+            # the per-file cap is checked against the declared size, again before
+            # the read, for the same reason as the total above.
+            if info.file_size > limits.max_osu_file_bytes:
+                raise OszArchiveError(
+                    f"difficulty {info.filename!r} is {info.file_size} bytes, "
+                    f"over the {limits.max_osu_file_bytes} byte limit",
+                )
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)